compressors/golomb_coder.py

```diff
@@ -0,0 +1,211 @@
+"""Golomb prefix coding for non-negative integers 
+
+Golomb code with parameter M based on
+https://en.wikipedia.org/wiki/Golomb_coding#Simple_algorithm.
+If M is power of 2, we simplify to Rice codes with slight
+change in logic.
+
+The main idea of Golomb code is to divide the input into quotient by M
```

Comment:
One small suggestion is to write this in points. Good that you included the decoding algo, as I didn't see that on the wikipedia link!